task/monorail-server/src/bsp/sidecar_bc.rs

```diff
@@ -447,14 +447,25 @@ impl<'a, R: Vsc7448Rw> Bsp<'a, R> {
             }
         }
 
-        // The 10G link should only be up if the host isn't trying to hold us in
-        // reset; we won't retrigger autonegotiation / system reset if that's
-        // not the case.
-        let check_10g_link = match self.seq.tofino_pcie_hotplug_status() {
-            Ok(pcie_hotplug_status) => {
-                let host_reset = pcie_hotplug_status & 1;
-                host_reset == 0
+        // The 10G link should only be up if we're in A0 and the host isn't
+        // trying to hold us in reset; we won't retrigger autonegotiation /
+        // system reset if that's not the case.
+        let check_10g_link = match self.seq.tofino_seq_state() {
+            Ok(TofinoSeqState::A0) => {
+                match self.seq.tofino_pcie_hotplug_status() {
+                    Ok(pcie_hotplug_status) => {
+                        let host_reset = pcie_hotplug_status & 1;
```

Comment:
This is fine for now, but if we're going to rely on this I should make this a proper (typed) part of the API. I'll do this next time I am working in the vicinity.